File: app/src/main/java/com/hezb/hplayer/base/BaseActivity.java
package com.hezb.hplayer.base;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;

/**
 * activity父类
 * Created by hezb on 2016/1/14.
 */
public abstract class BaseActivity extends FragmentActivity {

    protected Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = this;
        setContentView(getContentViewId());
    }

    protected abstract int getContentViewId();
}

File: app/src/main/java/com/hezb/hplayer/base/BaseActivity.kt
package com.hezb.hplayer.base

import android.content.Context
import androidx.appcompat.app.AppCompatActivity

/**
 * Project Name: HPlayer
 * File Name:    BaseActivity
 *
 * Description: Activity基类.
 *
 * @author  hezhubo
 * @date    2022年03月02日 21:30
 */
abstract class BaseActivity : AppCompatActivity() {

    private var isRunning: Boolean = false

    override fun onResume() {
        super.onResume()
        isRunning = true
    }

    override fun onPause() {
        super.onPause()
        isRunning = false
    }

    fun getContext(): Context {
        return this
    }

}